src/Component/spec/Metadata/OperationSpec.php

```diff
@@ -0,0 +1,566 @@
+    }
+
+    function it_could_have_a_template(): void
+    {
+        $this->withTemplate('book/show.html.twig')
+            ->getTemplate()
+            ->shouldReturn('book/show.html.twig')
+        ;
+    }
+
+    function it_has_no_repository_by_default(): void
+    {
+        $this->getRepository()->shouldReturn(null);
+    }
+
+    function it_could_have_a_repository_with_method(): void
+    {
+        $this->withRepository(['method' => 'createListQueryBuilder'])
+            ->getRepository()
+            ->shouldReturn(['method' => 'createListQueryBuilder'])
+        ;
+    }
+
+    function it_could_have_a_repository_with_method_and_arguments(): void
+    {
+        $this->withRepository(['method' => 'createListQueryBuilder', 'arguments' => ['foo' => 'fighters']])
+            ->getRepository()
+            ->shouldReturn(['method' => 'createListQueryBuilder', 'arguments' => ['foo' => 'fighters']])
+        ;
+    }
+
+    function it_has_no_criteria_by_default(): void
+    {
+        $this->getCriteria()->shouldReturn(null);
+    }
+
+    function it_could_have_criteria(): void
+    {
+        $this->withCriteria(['foo' => 'fighters'])
+            ->getCriteria()
+            ->shouldReturn(['foo' => 'fighters'])
+        ;
+    }
+
+    function it_has_no_requirements_by_default(): void
+    {
+        $this->getRequirements()->shouldReturn(null);
+    }
+
+    function it_could_have_requirements(): void
+    {
+        $this->withRequirements(['id' => '\d+'])
+            ->getRequirements()
+            ->shouldReturn(['id' => '\d+'])
+        ;
+    }
+
+    function it_has_no_options_by_default(): void
+    {
+        $this->getOptions()->shouldReturn(null);
+    }
+
+    function it_could_have_options(): void
+    {
+        $this->withOptions(['utf8' => true])
+            ->getOptions()
+            ->shouldReturn(['utf8' => true])
+        ;
+    }
+
+    function it_has_no_host_by_default(): void
+    {
+        $this->getHost()->shouldReturn(null);
+    }
+
+    function it_could_have_a_host(): void
+    {
+        $this->withHost('m.example.com')
+            ->getHost()
+            ->shouldReturn('m.example.com')
+        ;
+    }
+
+    function it_has_no_schemes_by_default(): void
+    {
+        $this->getSchemes()->shouldReturn(null);
+    }
+
+    function it_could_have_schemes(): void
+    {
+        $this->withSchemes(['https'])
+            ->getSchemes()
+            ->shouldReturn(['https'])
+        ;
+    }
+
+    function it_has_no_priority_by_default(): void
+    {
+        $this->getPriority()->shouldReturn(null);
+    }
+
+    function it_could_have_a_priority(): void
+    {
+        $this->withPriority(42)
+            ->getPriority()
+            ->shouldReturn(42)
+        ;
+    }
+
+    function it_has_no_vars_by_default(): void
+    {
+        $this->getVars()->shouldReturn(null);
+    }
+
+    function it_could_have_vars(): void
+    {
+        $this->withVars([
```

Comment:
maybe it could be better to name it templateVars, but it will be harder for conversions.